registry/server/api_integration_test.go

```diff
@@ -0,0 +1,147 @@
+// +build integration
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	pb "github.com/DataDog/kafka-kit/v3/registry/protos"
+)
+
+func TestCreateTopic(t *testing.T) {
+	s, err := testIntegrationServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := map[int]*pb.CreateTopicRequest{
+		// This should succeed.
+		0: &pb.CreateTopicRequest{
+			Topic: &pb.Topic{Name: "new_topic", Partitions: 1, Replication: 1},
+		},
+		// This should fail because we're trying to create the same topic twice.
+		1: &pb.CreateTopicRequest{
+			Topic: &pb.Topic{Name: "new_topic", Partitions: 1, Replication: 1},
+		},
+		// This should fail; incomplete request params.
+		2: &pb.CreateTopicRequest{
+			Topic: &pb.Topic{Name: "", Partitions: 1, Replication: 1},
+		},
+		// This should fail; incomplete request params.
+		3: &pb.CreateTopicRequest{},
+	}
+
+	expectedErrors := map[int]error{
+		0: nil,
+		1: ErrTopicAlreadyExists,
+		2: ErrTopicNameEmpty,
+		3: ErrTopicFieldMissing,
+	}
+
+	for i := 0; i < len(tests); i++ {
+		_, err := s.CreateTopic(context.Background(), tests[i])
+		if err != expectedErrors[i] {
+			t.Errorf("Expected error '%s' for test %d, got '%s'", expectedErrors[i], i, err)
+		}
+	}
+}
+
+func TestCreateTaggedTopic(t *testing.T) {
+	s, err := testIntegrationServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Tag a broker so we can test partition mapping by tag.
+	_, err = s.TagBroker(context.Background(), &pb.BrokerRequest{Id: 1001, Tag: []string{"key:value"}})
```

Comment:
I'm still trying to wrap my head around how context works. My understanding is that `Background()` here doesn't mean this is running async in the background, but jus that we are using passing the general background context to this function to run with, and it won't introduce race conditions - is that close?

Reply:
That is pretty much all correct; in testing we're using the default 'background' context which is mostly to pass in _any_ context as the required input param. In real life, the context will carry useful things like timeouts, cancellation functionality for the requestor, and other metadata such as the requestor IP (http transcoded requests are grpc under the hood, so there's also things like http headers passed into the context as well).